### PR TITLE
Harmony: MenuItem

### LINKED ISSUE #864: Harmony: MenuItem

### PR DESCRIPTION
**Pull request by DenisVorop**
## PR includes

- [x] Bug Fix
- [x] Feature

## Related issues

Resolve #864 
Related: 
- https://github.com/taskany-inc/issues/pull/2325

## QA Instructions, Screenshots, Recordings

https://github.com/taskany-inc/bricks/assets/95316053/31969d16-8de2-4ffa-89f6-7a0050a03e61

### CODE REVIEW

File: src/harmony/Input/Input.tsx [hunk 1]
```diff
@@ -61,10 +63,13 @@ export const Input = forwardRef<HTMLInputElement, InputProps>(
...
-        const wrapperStyles: Record<string, unknown> = useMemo(() => {
-            return {
-                '--pointer-events': pointerEvents || 'auto',
-            };
+        const [wrapperStyles, inputStyles]: Record<string, unknown>[] = useMemo(() => {
+            return [
+                {
+                    '--pointer-events': pointerEvents || 'auto',
+                },
+                { '--input-width': (typeof width === 'string' ? width : `${width}px`) || 'unset' },
```
[the remark this answers is not in the document]
Comment (DenisVorop) [reply src/harmony/Input/Input.tsx]: I used this approach to ensure that the cascade works. If the width is set as an inline style, it cannot be modified using the class

File: src/harmony/Input/Input.module.css [hunk 2]
```diff
@@ -112,7 +113,7 @@
 
     color: var(--text-ghost);
     transition: color 200ms ease-in-out 0s;
-    pointer-events: var(--pointer-events);
+    pointer-events: var(--pointer-events, auto);
```
Comment (awinogradov) [inline src/harmony/Input/Input.module.css:116]: `--input-pointer-events`

File: src/harmony/DatePicker/DatePicker.tsx [hunk 3]
```diff
@@ -766,12 +786,17 @@ export const DatePickerStrict: React.FC<DatePickerStrictProps> = ({
...
+                                            ? classes.DatePickerYearInput
+                                            : classes.DatePickerStrictDateInput,
+                                    )}
                                     type="number"
-                                    name={part}
+                                    name={inputNames['date-picker-strict-date'][part]}
                                     maxLength={valueLen[part]}
                                     value={dateValues[part]}
                                     onChange={handleChangeDatePart(part)}
+                                    width="100%"
```
Comment (awinogradov) [inline src/harmony/DatePicker/DatePicker.tsx:799]: Why not to do it via DatePicker css?

File: src/harmony/DatePicker/DatePicker.tsx [hunk 4]
```diff
@@ -355,6 +358,26 @@ const datePickerReducer: React.Reducer<DatePickerState, Actions> = (state, actio
...
 
+const quartersBricks = ['right', 'center', 'center', 'left'] as const;
+const aliasesBricks = ['right', 'center', 'left'] as const;
+const quarters: Quarter[] = ['Q1', 'Q2', 'Q3', 'Q4'] as const;
+const aliases: QuarterAlias[] = ['@prev', '@current', '@next'] as const;
+
+const inputNames = {
+    'date-picker-year': 'date-picker-year',
+    'date-picker-quarters': {
+        [quarters[0]]: `date-picker-quarter-${quarters[0]}`,
```
Comment (awinogradov) [inline src/harmony/DatePicker/DatePicker.tsx:369]: `Q1: 'date-picker-quarter-Q1'`

**Review by awinogradov — CHANGES_REQUESTED** [review]

**Review by awinogradov — APPROVED** [review]

**Review by asabotovich — APPROVED** [review]

**Pull request merged**